src/integrations/cleanup-integration.php

```diff
@@ -520,4 +533,48 @@ protected function cleanup_orphaned_from_table( $table, $column, $limit ) {
 		// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared, WordPress.DB.PreparedSQL.NotPrepared, WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching -- Reason: Already prepared.
 		return $wpdb->query( "DELETE FROM $table WHERE {$column} IN( " . \implode( ',', $orphans ) . ' )' );
 	}
+
+	/**
+	 * Updates the author_id of indexables which author_id is not in the wp_users table with the id of the reassingned user.
+	 *
+	 * @param int $limit The limit we'll apply to the queries.
+	 *
+	 * @return int|bool The number of updated rows, false if query to get data fails.
+	 */
+	protected function update_indexables_author_to_reassigned( $limit ) {
+		global $wpdb;
+
+		$indexable_table = Model::get_table_name( 'Indexable' );
+		$users_table     = $wpdb->users;
+		$posts_table     = $wpdb->posts;
+
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared -- Reason: There is no unescaped user input.
+		$query = $wpdb->prepare(
+			"
+			SELECT {$indexable_table}.object_id, {$indexable_table}.author_id, {$posts_table}.post_author
+			FROM {$indexable_table} JOIN {$posts_table} on {$indexable_table}.object_id =  {$posts_table}.id
+			WHERE object_type='post'
+			AND author_id NOT IN(SELECT id from {$users_table})
+			ORDER BY {$indexable_table}.author_id
+			LIMIT %d",
+			$limit
+		);
+		// phpcs:enable
+
+		// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared, WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching -- Reason: Already prepared.
+		$reassigned_authors = $wpdb->get_results( $query );
+
+		if ( $reassigned_authors === false ) {
+			return false;
+		}
+
+		foreach ( $reassigned_authors as $reassigned_author ) {
```

Comment:
I have a bit of a doubt about this part. If you have 20.000 posts that have a removed author you would loop 20.000 times here (or 1000 if you have the default limit) and just do the same thing over and over again. Would it be possible to make a big update query to update all posts to the author by first looping over the $reassigned_authors making an array with object_id's and then do that in 1 go?

Reply:
Done: it resulted in a bit more convoluted solution, but definitely more perfomant 😉